### PR TITLE
[HLS-2561] Sdk configs

### PR DESCRIPTION
**Pull request by nozik**

### CODE REVIEW

File: sdk/config.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	if samplerConfig.Key == "" {
+		return trace.AlwaysSample()
+	}
+
+	return trace.TraceIDRatioBased(samplerConfig.Value.AsFloat64())
+}
+
+func getCollectorEndpoint(collectorEndpointConfig attribute.KeyValue) string {
```
Comment (prsnca) [inline sdk/config.go:65]: these following function are doing exactly the same - can't we make them generic?

File: sdk/config_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	assert.Equal(t, config.serviceName, serviceName)
+	assert.Equal(t, config.apiToken, token)
+	assert.Equal(t, config.collectorEndpoint, defaultCollectorEndpoint)
+	assert.Equal(t, config.collectorPath, defaultCollectorPath)
+	assert.Equal(t, config.commitHash, "")
+	assert.Equal(t, config.environment, "")
+	assert.Equal(t, config.sampler.Description(), "AlwaysOnSampler")
+}
+
+func TestConfigWithOptions(t *testing.T) {
```
Comment (prsnca) [inline sdk/config_test.go:24]: can we test the env vars as well?

File: sdk/config.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	apiToken          string
+	sampler           trace.Sampler
+	collectorEndpoint string
+	collectorPath     string
+	environment       string
+	commitHash        string
+}
+
+// Keys and their matching env vars
+const samplingRatioKey = "samplingRatio"
```
Comment (prsnca) [inline sdk/config.go:22]: we can use `viper` to work with env vars and allow other configurations - but probably we should stay native

**Review by prsnca — CHANGES_REQUESTED** [review]

File: sdk/config.go (hunk 3 above)
Comment (nozik) [reply sdk/config.go:22]: I prefer staying native

File: sdk/config.go (hunk 1 above)
Comment (nozik) [reply sdk/config.go:65]: Done

File: sdk/config_test.go (hunk 2 above)
Comment (nozik) [reply sdk/config_test.go:25]: Done

**Review by prsnca — APPROVED** [review]

**Pull request merged**